### PR TITLE
adding ttHH HEFT cards at different values of kl,kt and c2

### PR DESCRIPTION
**Pull request by gekobs**
I am opening this PR to add the cards needed to produce ttHH events, varying the dim-6 HEFT couplings.
The generation is based on a UFO model provided by a group of theorist and it is not centrally available atm. Could you please advise on  where should I upload the UFO?
Thanks!

### CODE REVIEW

**Comment by menglu21** [comment]
Hi @gekobs you should upload the model to /afs/cern.ch/cms/generators/www, and you should have the permission if you are gen-contact.

File: bin/MadGraph5_aMCatNLO/cards/production/13TeV/ttHH_HEFT_LO/c2_1/ttHH_HEFT_c2_1_proc_card.dat [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+set complex_mass_scheme False
+#set include_lepton_initiated_processes False
+set gauge unitary
+set loop_optimized_output True
+set loop_color_flows False
+set max_npoint_for_channel 0
+set default_unset_couplings 99
+#set max_t_for_channel 99
+#set zerowidth_tchannel True
+#set nlo_mixed_expansion True
```
Comment (menglu21) [inline bin/MadGraph5_aMCatNLO/cards/production/13TeV/ttHH_HEFT_LO/c2_1/ttHH_HEFT_c2_1_proc_card.dat:35]: you can remove the banner and the default setup to make the card compact

File: bin/MadGraph5_aMCatNLO/cards/production/13TeV/ttHH_HEFT_LO/c2_1/ttHH_HEFT_c2_1_proc_card.dat [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+import model UFO_HEFT_h25-c2
+define p = g u c d s u~ c~ d~ s~
+define j = g u c d s u~ c~ d~ s~
+define l+ = e+ mu+
+define l- = e- mu-
+define vl = ve vm vt
+define vl~ = ve~ vm~ vt~
+define lep = e+ e- mu+ mu-
+define nus = ve~ vm~ ve vm
```
Comment (menglu21) [inline bin/MadGraph5_aMCatNLO/cards/production/13TeV/ttHH_HEFT_LO/c2_1/ttHH_HEFT_c2_1_proc_card.dat:44]: these lines seem to be not used, so you can remove them

File: bin/MadGraph5_aMCatNLO/cards/production/13TeV/ttHH_HEFT_LO/c2_1/ttHH_HEFT_c2_1_run_card.dat [hunk 3]
```diff
@@ -0,0 +1,268 @@
...
+#*********************************************************************
+# Beam polarization from -100 (left-handed) to 100 (right-handed)    *
+#*********************************************************************
+        0     = polbeam1 ! beam polarization for beam 1
+        0     = polbeam2 ! beam polarization for beam 2
+#*********************************************************************
+# PDF CHOICE: this automatically fixes also alpha_s and its evol.    *
+#*********************************************************************
+ 'lhapdf'    = pdlabel     ! PDF set                                  
+  $DEFAULT_PDF_SETS    = lhaid     ! if pdlabel=lhapdf, this is the lhapdf number
```
Comment (menglu21) [inline bin/MadGraph5_aMCatNLO/cards/production/13TeV/ttHH_HEFT_LO/c2_1/ttHH_HEFT_c2_1_run_card.dat:52]: please add "$DEFAULT_PDF_MEMBERS = reweight_PDF" after line 52

File: bin/MadGraph5_aMCatNLO/cards/production/13TeV/ttHH_HEFT_LO/c2_1/ttHH_HEFT_c2_1_run_card.dat [hunk 4]
```diff
@@ -0,0 +1,268 @@
...
+ 0   = xqcut   ! minimum kt jet measure between partons
+#*********************************************************************
+#
+#*********************************************************************
+# Store info for systematics studies                                 *
+# WARNING: If use_syst is T, matched Pythia output is                *
+#          meaningful ONLY if plotted taking matchscale              *
+#          reweighting into account!                                 *
+#*********************************************************************
+ F  = use_syst      ! Enable systematics studies
```
Comment (menglu21) [inline bin/MadGraph5_aMCatNLO/cards/production/13TeV/ttHH_HEFT_LO/c2_1/ttHH_HEFT_c2_1_run_card.dat:268]: please use "T  = use_syst"

**Review by menglu21 — COMMENTED** [review]
I have several general comments:
1. if the difference among all those processes are the coupling, i suggest you can use template cards for one process and use script to generate cards for other couplings
2. is the parameter card the same with the default one, if yes, you can delete the parameter card. Basically we suggest to put all the modification of model in the customized card
3. i guess those cards are for UL, if yes, please push them to [genproductions](https://github.com/cms-sw/genproductions)/[bin](https://github.com/cms-sw/genproductions/tree/master/bin)/[MadGraph5_aMCatNLO](https://github.com/cms-sw/genproductions/tree/master/bin/MadGraph5_aMCatNLO)/[cards](https://github.com/cms-sw/genproductions/tree/master/bin/MadGraph5_aMCatNLO/cards)/[production](https://github.com/cms-sw/genproductions/tree/master/bin/MadGraph5_aMCatNLO/cards/production)/[2017](https://github.com/cms-sw/genproductions/tree/master/bin/MadGraph5_aMCatNLO/cards/production/2017)/13TeV/

**Comment by gekobs** [comment]
> Hi @gekobs you should upload the model to /afs/cern.ch/cms/generators/www, and you should have the permission if you are gen-contact.

Dear Meng, thanks much for your comments, I will implement your suggestions and update the PR. Regarding being a gen-contact l, though, I am not. As my analysis is under Hgg, should I ask the Hgg gen contact to do that on my behalf or can I ask for permissions somewhere? 
Thanks!

**Comment by menglu21** [comment]
> > Hi @gekobs you should upload the model to /afs/cern.ch/cms/generators/www, and you should have the permission if you are gen-contact.
> 
> Dear Meng, thanks much for your comments, I will implement your suggestions and update the PR. Regarding being a gen-contact, though, I am not. As my analysis is under Hgg, should I ask the Hgg gen contact to do that on my behalf or can I ask for permissions somewhere? Thanks!

yes, please ask the mc contact to upload the model.

**Comment by gekobs** [comment]
Hi @menglu21 ,
I pushed a commit that should address your comments.
I am not too sure regarding the script you mention in your first comment though. The way I generate the cards on my end is indeed a bash script that creates a coupling folder, copy there the base cards changing the names and update the cards content accordingly. Do you want me to push this? or are you referring to some central script I am not aware of?
Thanks!

**Comment by menglu21** [comment]
> 

yes, I was saying the bash script, that will make our review more efficiently, thanks

**Comment by gekobs** [comment]
Dear @menglu21, I edited again the PR to including the script you requested and removing the multiple copies of cards.  Can you please proceed with the review of the PR?
Thanks! Giacomo

**Comment by menglu21** [comment]
cards look good to me, @Saptaparna @mlizzo @Dominic-Stafford please have a check and feel free to merge

**Comment by Saptaparna** [comment]
Looks fine to me. @gekobs: just one thing, so you know, typically people produce events like this:

- `generate p p > t t~ h h` with an extra jet (`add process p p > t t~ h h j`) and merge and match it 

Your process looks fine as long as you know that you are doing everything at LO and while the inclusion of the extra jet does not actually help with that, it is still a proxy for the NLO generation to some extent.

**Pull request merged**